### PR TITLE
Use Yarn Workspaces

### PR DESCRIPTION
**Pull request by blink1073**
Switches over to using yarn workspaces following https://yarnpkg.com/blog/2017/08/02/introducing-workspaces/.  cf https://github.com/jupyterlab/jupyterlab/pull/3168

```bash
$ git clean -dfx && yarn cache clean
yarn cache v1.2.1
success Cleared cache.
✨  Done in 0.04s.

$ yarn
yarn install v1.2.1
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "postcss-cssnext@3.0.2" has unmet peer dependency "caniuse-lite@^1.0.30000697".
warning "karma-chai@0.1.0" has unmet peer dependency "chai@*".
warning "karma-chai@0.1.0" has unmet peer dependency "karma@>=0.10.9".
[4/4] 📃  Building fresh packages...
✨  Done in 37.08s.

$ yarn
yarn install v1.2.1
[1/4] 🔍  Resolving packages...
success Already up-to-date.
✨  Done in 0.76s.

$ git clean -dfx && yarn
Removing examples/web-tmpnb/node_modules/
Removing examples/web1/node_modules/
Removing examples/web2/node_modules/
Removing examples/web3/node_modules/
Removing examples/web4/node_modules/
Removing node_modules/
Removing packages/base/node_modules/
Removing packages/controls/node_modules/
Removing packages/html-manager/node_modules/
Removing packages/jupyterlab-manager/node_modules/
Removing packages/output/node_modules/
Removing widgetsnbextension/node_modules/
yarn install v1.2.1
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "postcss-cssnext@3.0.2" has unmet peer dependency "caniuse-lite@^1.0.30000697".
warning "karma-chai@0.1.0" has unmet peer dependency "chai@*".
warning "karma-chai@0.1.0" has unmet peer dependency "karma@>=0.10.9".
[4/4] 📃  Building fresh packages...
✨  Done in 13.86s.
```

### CODE REVIEW

**Comment by blink1073** [comment]
cc @rgbkrk

**Comment by jasongrout** [comment]
Thank you *very* much!

**Comment by rgbkrk** [comment]
Nice! I’ve been hoping to switch nteract’s, it requires more fix ups to electron builder.

File: package.json [hunk 1]
```diff
@@ -12,6 +11,11 @@
     "publish": "npm run clean && npm run build && lerna publish -m \"Publish npm packages\""
```
Comment (jasongrout) [inline package.json]: Why do these scripts still use `npm run clean`, `npm run build`? Should they be changed to `yarn run clean` or `yarn run build`?

**Comment by jasongrout** [comment]
Other than the one inline clarifying comment above, this looks good to me. Thanks!

**Comment by jasongrout** [comment]
@blink1073 - I'm now running into https://github.com/lerna/lerna/issues/688 when trying to publish using yarn. Have you encountered that?

**Comment by jasongrout** [comment]
At least one problem was solved by setting the registry in lerna.json (see https://github.com/lerna/lerna/issues/896)

**Comment by jasongrout** [comment]
It worked with the above fix! I also cleaned out the yarn cache and reset the lock file to use npmjs.org instead of yarnpkg.com.

**Comment by blink1073** [comment]
Great!  I'll apply the fix to the other PRs.

**Comment by blink1073** [comment]
Did you apply a registry config to yarn?  If so that should be put in a .yarnrc file.

**Comment by jasongrout** [comment]
Yes, I put it in my user .yarnrc, but should have put it in the repo for others.

**Comment by jasongrout** [comment]
Fixed in https://github.com/jupyter-widgets/ipywidgets/pull/1821

**Pull request merged**